### PR TITLE
Add test for handling NULL values in PostgreSQL connector

### PR DESCRIPTION
**Pull request by strk**
See https://github.com/qgis/QGIS/pull/33225#issuecomment-564446756

### CODE REVIEW

File: tests/src/python/test_qgsproviderconnection_postgres.py [hunk 1]
```diff
@@ -200,6 +200,15 @@ def test_true_false(self):
         self.assertEqual(conn.executeSql('SELECT FALSE'), [[False]])
         self.assertEqual(conn.executeSql('SELECT TRUE'), [[True]])
 
+    def test_true_false(self):
```
Comment (elpaso) [inline tests/src/python/test_qgsproviderconnection_postgres.py]:
```suggestion
    def test_nulls(self):
```

**Review by elpaso — CHANGES_REQUESTED** [review]
change the test name or it will shadow the previous one

**Comment by strk** [comment]
OOps, force-pushed a fix

**Review by elpaso — APPROVED** [review]

**Comment by strk** [comment]
@elpaso the tests indeed fail:
https://travis-ci.org/qgis/QGIS/jobs/623752437#L2455-L2471

Those NULL values are returned as False instead of None

**Comment by strk** [comment]
(but only for the bool case?)

**Comment by strk** [comment]
Is it ok to merge with a failing Azure pipeline ? (seems to be)
